rust/noosphere-gateway/src/route/replicate.rs

```diff
@@ -62,5 +73,302 @@ where
         Duration::from_secs(10),
     );
 
-    Ok(StreamBody::new(car_stream(store, memo_version)))
+    if let Some(since) = since {
+        let since_memo = store
+            .load::<DagCborCodec, MemoIpld>(&since)
+            .await
+            .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)?;
+        let latest_memo = store
+            .load::<DagCborCodec, MemoIpld>(&memo_version)
+            .await
+            .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)?;
+
+        if since_memo.lamport_order() < latest_memo.lamport_order() {
+            // We should attempt to replicate incrementally
+            if is_allowed_to_replicate_incrementally(&since_memo, &latest_memo) {
+                // TODO(#408): To mitigate abuse, we should probably cap the spread
+                // between revisions to some finite value that is permissive of 99%
+                // of usage. Maybe somewhere in the ballpark of 1~10k revisions. It
+                // should be a large-but-finite number.
+                debug!("Streaming revisions from {} to {}", since, memo_version);
+                return Ok(StreamBody::new(Box::pin(car_stream(
+                    vec![memo_version],
+                    memo_history_stream(store, &memo_version.into(), Some(&since)),
+                ))));
+            } else {
+                error!("Suggested version {since} is not a valid ancestor of {memo_version}");
+                return Err(StatusCode::BAD_REQUEST);
+            }
+        } else {
+            warn!(
```

Comment:
If `since_memo == latest_memo`, there should be nothing to sync I believe, which triggers this slightly inaccurate warning

Reply:
Tweaked `is newer` to `is not older`; burden is on the caller to provide expected inputs in this case, so it's correct to warn.